File: tree/binary_tree/c++/inorder_traversal.cpp
#include <vector>

using namespace std;

/*** Definition for a binary tree node. */
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    void traverse(TreeNode* node, vector<int> *vals){
        if (node != NULL){
            traverse(node->left, vals);
            vals->push_back(node->val);
            traverse(node->right, vals);
        }
        return;     // very important to increase the performance
    }

    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> res;
        traverse(root, &res);
        return res;
    }
};

File: tree/binary_tree/python/inorder_traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root):
        def traverse(node, vals):
            if node:
                traverse(node.left, vals)
                vals.append(node.val)
                traverse(node.right, vals)
        res = []
        traverse(root, res)
        return res